core/src/main/scala/cats/data/Validated.scala

```diff
@@ -374,6 +375,20 @@ private[data] sealed abstract class ValidatedInstances1 extends ValidatedInstanc
       def combine(x: Validated[A, B], y: Validated[A, B]): Validated[A, B] = x combine y
     }
 
+  implicit def catsDataCommutativeApplicativeForValidated[E: CommutativeSemigroup]: CommutativeApplicative[Validated[E, ?]] =
```

Comment:
Is it possible to extract this and `ApplicativeError` to a common super trait?

Reply:
You mean adding a `CommutativeApplicativeError` typeclass? Or just reshuffling the instances?

Reply:
not a new type class, just a new private trait for duplicated code in the two instances.

Reply:
Ah of course, will do! :)